coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -0,0 +1,55 @@
+package filodb.coordinator.queryplanner
+
+import filodb.core.metadata.Dataset
+import filodb.core.query.{Filter, QueryContext}
+import filodb.query.{Aggregate, BinaryJoin, LogicalPlan}
+import filodb.query.exec.{DistConcatExec, ExecPlan, InProcessPlanDispatcher, ReduceAggregateExec}
+
+case class ShardColumnValues(regexColumn: String, regexValue: String, nonRegexColumn: String, nonRegexValue: String)
+
+/**
+  * Responsible for query planning for queries having regex in shard column
+  *
+  * @param nonRegexShardColumn shard column which is mandatory and cannot be a regex
+  * @param dataset dataset
+  * @param multiPartitionPlanner multiPartition query planner
+  * @param regexFieldMatcher used to get values for nonRegexShardColumn matching shard key column regex.
+  */
+
+class ShardKeyRegexPlanner(nonRegexShardColumn: String,
+                           dataset: Dataset,
+                           multiPartitionPlanner: QueryPlanner,
+                           regexFieldMatcher: ShardColumnValues => Seq[String]) extends QueryPlanner {
```

Comment:
Have you considered using the In operator of ColumnFilter instead of using Regex representation in the logical plan?
Also I am unclear on the design of `regexFieldMatcher` function.

Reply:
regexFieldMatcher will return all values for ShardColumnValues.regexColumn(_ns_) matching ShardColumnValues .regexValue for ShardColumnValues. nonRegexColumn(_ws_)